python/examples/integrations/bentoml/README.md

```diff
@@ -0,0 +1,188 @@
+
+## 🛠️ How to implement ML monitoring for BentoML with whylogs and WhyLabs:
+
+After running the training script with `python train.py`, you'll see a message similar to "`Model saved: Model(tag="iris_knn:fhfsvifgrsrtjlg6")`". This is the versioned model saved to the BentoML local model store.
+
+> Note: It can be good practice to log the training data as a static reference profile to use as a baseline to detect data drift. For more information, check out the [whylogs documentation](https://whylogs.readthedocs.io/en/latest/examples/integrations/writers/Writing_Reference_Profiles_to_WhyLabs.html).
+
+Next, In `service.py`, we'll get ready to serve and monitor our latest saved model.
+
+To get the WhyLabs API key and org-id, go to the [WhyLabs dashboard](https://hub.whylabsapp.com/) and click on the "Access Tokens" section in the settings menu. Model-ids are found under the "Project Management" tab in the same section.
+
+```python
+import atexit
+import os
+
+import bentoml
+import numpy as np
+from bentoml.io import NumpyNdarray, Text
+
+import whylogs as why
+
+# Set WhyLabs environment variables
+os.environ["WHYLABS_API_KEY"] = "APIKEY"
+os.environ["WHYLABS_DEFAULT_ORG_ID"] = "ORGID"
+os.environ["WHYLABS_DEFAULT_DATASET_ID"] = "PROJECTID"
+
+# Model Prediction Class Names
+CLASS_NAMES = ["setosa", "versicolor", "virginica"]
+
+# Load model from BentoML local model store & create a BentoML runner
+iris_clf_runner = bentoml.sklearn.get("iris_knn:latest").to_runner()
+svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
+
+```
+
+> Note: A best practice for software deployment is to have these environment variables always set on the machine/environment level (such as per the CI/QA machine, a Kubernetes Pod, etc.) so it is possible to leave code untouched and persist it correctly according to the environment you execute it.
+
+Now that our bentoML runner and environment variables are set, we can create an API endpoint with BentoML and set up model and data monitoring with whylogs and WhyLabs. We'll Log the data, predicted class, and model probability score.
+
+Create a class in `service.py` to handle startup, shutdown, logging, and prediction logic with Bentoml. This example uses the [rolling log](https://whylogs.readthedocs.io/en/latest/examples/advanced/Log_Rotation_for_Streaming_Data/Streaming_Data_with_Log_Rotation.html) functionality of whylogs to write profiles to WhyLabs at set intervals. The `atexit` module registers a callback function to close the whylogs rolling logger when the service is shut down.
+
+```python
+# create a class to handle startup, shutdown, and prediction logic with Bentoml
+class MyService:
+    def __init__(self):
+        # Initialize the whylogs rolling logger with a 5 minute interval
+        global logger
+        logger = why.logger(
+            mode="rolling", interval=5, when="M", base_name="bentoml_predictions"
+        )
+        logger.append_writer("whylabs")
```

Comment:
Hi @sagecodes, we recently introduced the `on_startup` lifecycle hook, which should make the logger initialization a lot cleaner to implement: https://docs.bentoml.org/en/latest/concepts/service.html#lifecycle-hooks